### PR TITLE
Adds documentation for HiveServer2 support

### PR DESCRIPTION
**Pull request by linghengqian**
For #29052.

Changes proposed in this pull request:
  - Adds documentation for HiveServer2 support.
  - TL;DR, no developer would like to see a long discussion about HiveServer2 in the documentation of GraalVM Native Image integration. **It is necessary to separate the independent documentation of HiveServer2.**
  - Further improve the integration documentation of Testcontainers Java.
  - Bump the MySQL Docker Image used by nativeTest to `mysql:9.1.0-oraclelinux9`.
  - Bump the MS SQL Server Docker Image used by nativeTest to `mcr.microsoft.com/mssql/server:2022-CU16-ubuntu-22.04`.
  - Bump the Postgres Docker Image used by nativeTest to `postgres:17.1-bookworm`.
  - Bump the Clickhouse Docker Image used by nativeTest to `clickhouse/clickhouse-server:24.10.2.80`.

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [x] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.
- [x] I have updated the Release Notes of the current development version. For more details, see [Update Release Note](https://shardingsphere.apache.org/community/en/involved/contribute/contributor/)

### CODE REVIEW

**Review by linghengqian — COMMENTED** [review]
From https://github.com/apache/shardingsphere/actions/runs/11908587236/job/33188321588, the performance of github actions equipment is too poor, and it can't even guarantee the weak consistency of hiveserver2 transactions. Let me see what I can do.
